src/connection/DbConnection.js

```diff
@@ -167,6 +168,67 @@ module.exports = (
         });
     }
 
+    async function transactionQuery(sqls, multiValues = [], label) {
+        const outputLabel = label || DEFAULT_OUTPUT_LABEL;
+        const startToken = timers.start();
+
+        let connection;
+        try {
+            connection = await newConnection();
+        } catch (err) {
+            logger.error(`${outputLabel}.sql.connectionFailure`, err);
+            connection.destroy();
+            throw err;
+        }
+
+        // Promisified methods. Must be attached to keep 'this' context.
+        ['beginTransaction', 'query', 'rollback', 'commit'].forEach((method) => {
+            connection[`${method}Promise`] =
+                util.promisify(connection[method]);
+        });
+
+        try {
```

Comment:
I'd rather use `try-catch-finally` throughout this function so that we can guarantee the connection will be destroyed.

```
try {
    // do stuff
} catch (err) {
    // do stuff
} finally {
    connection.destroy();
}
```